File: dart/behavioral/visitor/guru_example/bin/concrete_visitor1.dart
import 'concrete_component_a.dart';
import 'concrete_component_b.dart';
import 'visitor.dart';

class ConcreteVisitor1 implements Visitor {
  @override
  void visitConcreteComponentA(ConcreteComponentA element) {
    print('ConcreteVisitor1: visitConcreteComponentA ${element.a()}.');
  }

  @override
  void visitConcreteComponentB(ConcreteComponentB element) {
    print('ConcreteVisitor1: visitConcreteComponentB ${element.b()}.');
  }
}

File: typescript/behavioral/visitor/guru_example/concreteVisitor1.ts
import {Visitor} from "./visitor";
import {ConcreteComponentA} from "./concreteComponentA";
import {ConcreteComponentB} from "./concreteComponentB";

export class ConcreteVisitor1 implements Visitor {
  public visitConcreteComponentA(element: ConcreteComponentA): void {
    console.log(`ConcreteVisitor1: visitConcreteComponentA ${element.exclusiveMethodOfConcreteComponentA()}.`)
  }

  public visitConcreteComponentB(element: ConcreteComponentB): void {
    console.log(`ConcreteVisitor2: visitConcreteComponentB ${element.specialMethodOfConcreteComponentB()}.`)
  }
}
